collated/main/A0111277M.md

```diff
@@ -0,0 +1,594 @@
+            }
+            
+            if (!dateMap.containsKey(END_DATE)) {
+                dateMap.put(END_DATE, ( (Event) taskToEdit).getEndDate()) ;
+            } else {
+                hasChanged = true ;
+            }
+
+        }
+
+        if (taskToEdit instanceof Deadline) {
+
+            if (!dateMap.containsKey(END_DATE)) {
+                dateMap.put(END_DATE, ( (Deadline) taskToEdit).getEndDate()) ;
+            } else {
+                hasChanged = true ;
+            }
+        }
+        
+        if(dateMap.containsKey(END_DATE)) {
+            if (dateMap.get(END_DATE).equals(DateUtil.MARKER_FOR_DELETE)) {
+            	dateMap.put(END_DATE, null);
+            } 
+        } else {
+        	dateMap.put(END_DATE, null);
+        }
+
+        
+        if (dateMap.containsKey(START_DATE)) {
+        	if (dateMap.get(START_DATE).equals(DateUtil.MARKER_FOR_DELETE)) {
+        		dateMap.put(START_DATE, null);
+        	} 
+        } else {
+    		dateMap.put(START_DATE, null);
+    	}
+        
+    
+    }
+    
+
+    /* 
+     * Updates the taglist
+     */
+    private Set<Tag> editOrDeleteTags(UniqueTagList currentTags, Set<String> tagNamesSet) throws IllegalValueException {
+        
+        assert model != null ;
+        
+        Set<Tag> newTaskTags = Sets.newHashSet(currentTags) ;
+        
+        for (String names : tagNamesSet) {
+            
+            Tag thisTag = model.getTagRegistry().getTagFromString(names, true) ;
+            
+            if (!currentTags.contains(thisTag)) {
+                newTaskTags.add(thisTag) ;
+                hasChanged = true ;
+                
+            } else {
+                newTaskTags.remove(thisTag) ;
+                hasChanged = true ;
+            }
+            
+        }
+        
+        return newTaskTags ;
+    }
+
+    @Override
+    public Pair<List<ReadOnlyTask>, List<ReadOnlyTask>> getCommandChanges() {
+        return new Pair<List<ReadOnlyTask>, List<ReadOnlyTask>>(ImmutableList.copyOf(tasksAdded), ImmutableList.copyOf(tasksDeleted)) ; 
+    }
+}
+```
+###### \java\seedu\address\logic\commands\MarkCommand.java
+``` java
+ */
+public class MarkCommand extends Command {
+
+	public static final String COMMAND_WORD = "mark";
+	
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Mark the task identified by the index number used in the last task listing as done.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_MARK_TASK_SUCCESS_DONE = "Marked Task : [%1$s] as done - ";
+    public static final String MESSAGE_MARK_TASK_SUCCESS_UNDONE = "Marked Task : [%1$s] as undone - ";
+    
+    public final int targetIndex;
+    private final List<ReadOnlyTask> tasksAdded = Lists.newLinkedList();
+    private final List<ReadOnlyTask> tasksDeleted = Lists.newLinkedList();
+    public MarkCommand(int targetIndex) {
```

Comment:
Need a line gap here You have unnecessary line gaps and not  in required places. So need attention. This is a serious issue in terms of code quality